### DISCUSSION TITLE
RPM packaged configuration file for ostree remote repositories

### DISCUSSION THREAD

**Discussion by pcdubs** [ostreedev/ostree#3079] [General]
In Fedora IoT we have been trying to use a drop in config file[1] to set the ostree remote details to use for updates. We were hoping this would be as easy as creating an RPM to install the file '/etc/ostree/remotes.d/fedora-iot.conf'. 

Unfortunately, this doesn't seem to work and still needs some of the commands we previously used in the kickstart[2] to move from the installation repository to the Fedora IoT repository listed in the config file. 

Is it possible to use an RPM packaged config file as we're attempting, without running any other commands on the host? 

[1] - https://src.fedoraproject.org/rpms/fedora-iot-config/blob/rawhide/f/fedora-iot.conf
[2] - https://pagure.io/fedora-kickstarts/blob/main/f/fedora-iot.ks#_37

**Comment by travier** [comment]
> Unfortunately, this doesn't seem to work

Can you clarify what does not work exactly?

> still needs some of the commands we previously used in the kickstart[2]

Is removing the commands from line 37 to 49 what you want to do?

**Reply by travier** [reply]
Reading the kickstart again, I would expect removing line 39 to 49 to work.

**Reply by travier** [reply]
https://src.fedoraproject.org/rpms/fedora-iot-config/blob/rawhide/f/fedora-iot.conf#_3 says false and https://pagure.io/fedora-kickstarts/blob/main/f/fedora-iot.ks#_49 is true 🤔

**Reply by nullr0ute** [reply]
so the problem is if we just drop a file in /etc/ostree/remotes.d/ it's not picked up as a remote. We have to run commands to do that which means we can't automatically configure that at build time if we're just building an ostree and shoving it in a bucket (VM/raw/other image).

**Comment by cgwalters** [comment]
I think we should have first-class kickstart support for this.  It might be just something like `ostreesetup --set-target-remote fedora-iot` or maybe just `ostreesetup --set-default-target-remote` that just finds the first remote in `/etc/ostree/remotes.d`.

**Reply by cgwalters** [reply]
Hmm though I see the problem; it's annoying to duplicate the URL right?

Maybe what would be the most ergonomic is something like:

```
ostreesetup --nogpg --osname=fedora-iot --remote=fedora-iot --source-url=https://kojipkgs.fedoraproject.org/compose/iot/repo/ --ref=fedora/rawhide/${basearch}/iot
```

(Note s/--url/--source-url/ !)  The semantics here would be that the remote would only be used to fetch.  After that, Anaconda would change the origin to match the remote it found in `/etc` from the deployed OS tree.

**Comment by pcdubs** [comment]
Adding ```--remote fedora-iot``` to the ```ostreesetup``` line in the kickstart created by osbuild was the part we were missing for it to use the packaged config file in the ostree rather than the installation source with the Anaconda installer. This is now working as intended, thank you for your help.
